operators/src/processing/bandwise_expression/mod.rs

```diff
@@ -0,0 +1,438 @@
+            source,
+            expression,
+            map_no_data: self.params.map_no_data,
+        }))
+    }
+
+    span_fn!(BandwiseExpression);
+}
+
+pub struct InitializedBandwiseExpression {
+    name: CanonicOperatorName,
+    result_descriptor: RasterResultDescriptor,
+    source: Box<dyn InitializedRasterOperator>,
+    expression: ExpressionAst,
+    map_no_data: bool,
+}
+
+impl InitializedRasterOperator for InitializedBandwiseExpression {
+    fn result_descriptor(&self) -> &RasterResultDescriptor {
+        &self.result_descriptor
+    }
+
+    fn query_processor(&self) -> Result<TypedRasterQueryProcessor> {
+        let typed_raster_processor = self.source.query_processor()?.into_f64();
+
+        let output_type = self.result_descriptor().data_type;
+
+        // TODO: spawn a blocking task for the compilation process
+        let expression_dependencies = get_expression_dependencies()
+            .map_err(|source| RasterExpressionError::Dependencies { source })?;
+
+        let expression = LinkedExpression::new(
+            self.expression.name(),
+            &self.expression.code(),
+            expression_dependencies,
+        )
+        .map_err(RasterExpressionError::from)?;
+
+        Ok(call_generic_raster_processor!(
+            output_type,
+            BandwiseExpressionProcessor::new(
+                typed_raster_processor,
+                self.result_descriptor.clone(),
+                expression,
+                self.map_no_data
+            )
+            .boxed()
+        ))
+    }
+
+    fn canonic_name(&self) -> CanonicOperatorName {
+        self.name.clone()
+    }
+}
+
+pub(crate) struct BandwiseExpressionProcessor<TO> {
+    source: Box<dyn RasterQueryProcessor<RasterType = f64>>,
+    result_descriptor: RasterResultDescriptor,
+    expression: Arc<LinkedExpression>,
+    map_no_data: bool,
+    phantom: std::marker::PhantomData<TO>,
+}
+
+impl<TO> BandwiseExpressionProcessor<TO>
+where
+    TO: Pixel,
+{
+    pub fn new(
+        source: Box<dyn RasterQueryProcessor<RasterType = f64>>,
+        result_descriptor: RasterResultDescriptor,
+        expression: LinkedExpression,
+        map_no_data: bool,
+    ) -> Self {
+        Self {
+            source,
+            result_descriptor,
+            expression: Arc::new(expression),
+            map_no_data,
+            phantom: Default::default(),
+        }
+    }
+
+    #[inline]
+    fn compute_expression(
+        raster: RasterTile2D<f64>,
+        expression: &LinkedExpression,
+        map_no_data: bool,
+    ) -> Result<GridOrEmpty2D<TO>> {
+        let expression = unsafe {
+            // we have to "trust" that the function has the signature we expect
+            expression
+                .function_1::<Option<f64>>()
+                .map_err(RasterExpressionError::from)?
+        };
+
+        let map_fn = |in_value: Option<f64>| {
+            // TODO: could be a |in_value: T1| if map no data is false!
```

Comment:
why only if no data is false? no data is a mask so id doesnt matter if it is a float type?

Reply:
if we knew map_no_data is false the function from the expression does not need to be able to handle no data values and thus would not need to have an Option as input. I just copied the code inclusive comment from the other expression operator. This could just be an optimization.